src/porepy/applications/verification_setups/manu_poromech_nofrac.py

```diff
@@ -362,159 +466,162 @@ def flow_source(self, sd: pp.Grid, time: float) -> np.ndarray:
             grid=sd,
-            true_array=self.exact_displacement,
-            approx_array=self.approx_displacement,
+            true_array=out.exact_displacement,
+            approx_array=out.approx_displacement,
             is_scalar=False,
             is_cc=True,
         )
 
-        # Poroelastic force
-        self.exact_force = ex.poroelastic_force(sd=sd, time=t)
-        self.approx_force = data[pp.STATE][pp.ITERATE][force_name]
-        self.error_force = self.relative_l2_error(
+        # Flux
+        out.exact_flux = self.exact_sol.darcy_flux(sd=sd, time=t)
+        out.approx_flux = data[pp.STATE][pp.ITERATE][q_name]
+        out.error_flux = self.relative_l2_error(
             grid=sd,
-            true_array=self.exact_force,
-            approx_array=self.approx_force,
+            true_array=out.exact_flux,
+            approx_array=out.approx_flux,
+            is_scalar=True,
+            is_cc=False,
+        )
+
+        # Force
+        out.exact_force = self.exact_sol.poroelastic_force(sd=sd, time=t)
+        out.approx_force = data[pp.STATE][pp.ITERATE][force_name]
+        out.error_force = self.relative_l2_error(
+            grid=sd,
+            true_array=out.exact_force,
+            approx_array=out.approx_force,
             is_scalar=False,
             is_cc=False,
         )
 
+        return out
+
+
+class SetupUtilities:
+    """Mixin class containing useful utility methods for the setup."""
+
+    mdg: pp.MixedDimensionalGrid
+    """Mixed-dimensional grid."""
+
+    results: list[SaveData]
+    """List of SaveData objects."""
+
+    def plot_results(self) -> None:
+        """Plotting results."""
+        self._plot_displacement()
+        self._plot_pressure()
+
+    def _plot_pressure(self):
+        """Plot exact and numerical pressures."""
+
+        sd = self.mdg.subdomains()[0]
+        p_ex = self.results[-1].exact_pressure
+        p_num = self.results[-1].approx_pressure
+
+        pp.plot_grid(sd, p_ex, plot_2d=True, linewidth=0, title="p (Exact)")
+        pp.plot_grid(sd, p_num, plot_2d=True, linewidth=0, title="p (Numerical)")
 
-# ----> Simulation model
+    def _plot_displacement(self):
+        """Plot exact and numerical displacements."""
+
+        sd = self.mdg.subdomains()[0]
+        u_ex = self.results[-1].exact_displacement
+        u_num = self.results[-1].approx_displacement
 
-# --------> Geometry
-class UnitSquare(pp.ModelGeometry):
+        # Horizontal displacement
+        pp.plot_grid(sd, u_ex[::2], plot_2d=True, linewidth=0, title="u_x (Exact)")
+        pp.plot_grid(sd, u_num[::2], plot_2d=True, linewidth=0, title="u_x (Numerical)")
+
+        # Vertical displacement
+        pp.plot_grid(sd, u_ex[1::2], plot_2d=True, linewidth=0, title="u_y (Exact)")
+        pp.plot_grid(
+            sd, u_num[1::2], plot_2d=True, linewidth=0, title="u_y (Numerical)"
+        )
+
+
+class UnitSquareTriangleGrid(pp.ModelGeometry):
```

Comment:
The Geometry classes should be refactored to a applications/geometries. This will also reveal potential for reuse, I think.

Reply:
I agree. Shall we postpone this until we have a clear plan on how to implement this? Adaptation should be straightforward for the verification setups.

Reply:
I would vote for doing it straight away. But I guess I could be convinced to postpone.